sql/layers/layers.go

```diff
@@ -64,6 +65,47 @@ func SetHareOutput(db sql.Executor, lid types.LayerID, output types.BlockID) err
 	return nil
 }
 
+// SetHareOutputWithCert sets the hare output for the layer with a block certificate.
+func SetHareOutputWithCert(db sql.Executor, lid types.LayerID, cert *types.Certificate) error {
+	output := cert.BlockID
+	data, err := codec.Encode(cert)
+	if err != nil {
+		return fmt.Errorf("encode cert %w", err)
+	}
+	if _, err := db.Exec(`insert into layers (id, hare_output, cert) values (?1, ?2, ?3) 
+					on conflict(id) do update set hare_output=?2, cert=?3;`,
```

Comment:
no. in a steady state. we will always first call SetHareOutput (by mesh.go after a block is generated)
this is important so tortoise can make progress right away without waiting for the certificate.

later, when a block is certified, then it call SetHareOutputWithCert to store the cert. it can change the local opinion at this point. for example, if the local hare fails, it will call SetHareOutput() with types.EmptyBlockID. but later when we sync Certificate from a peer, we then can overwrite this to their hare output.

i don't want to put application logic here. a certificate can only be saved if it gathered f+1 signatures. any certificate that passed that threshold can be saved here.

Reply:
Okay, so we need to support multiple blocks per layer. I think we need to move the certificates to the blocks table.
@noamnelke says it's rare, but we need to support multiple blocks per layer. I don't think we can just overwrite previous opinions.

Reply:
even if we want to accept multiple blocks per layer, can you explain how we can handle when hare output EmptyBlockID if we use blocks table?

Reply:
I think blockIDs should still be immutable.

Reply:
i don't think you are understanding my question. if we put the certificate in table "blocks", where will i save certificate for an empty block id from layer 100, 111, 123? they are layers that hare output empty block

Reply:
> In the situation you describe, there would be either one or 0 hare outputs --- once the rejoin occurs, the previously certified hare output might end up decertified, since the threshold has risen

you can have multiple certificates when the part of the same committee signs more than one block on purpose. in such case we should vote for empty layer. in order for this to happen you need an isolated partition where only malicious nodes can communicate with both sides, and maybe even more requirements

if committee changes based on the atxs state - previous certificate may gen invalidated, but in such case you will still have 1 or 0.

Reply:
i'll do a follow up PR once research conclude on this.

Reply:
concluded as in https://github.com/spacemeshos/go-spacemesh/pull/3449#discussion_r949231928

Reply:
@countvonzero that comment has nothing to do with the comment above

Reply:
i stand corrected. it should be this one that concludes this. https://github.com/spacemeshos/go-spacemesh/pull/3449#discussion_r949139516

do you agree? @dshulyak

Reply:
my point was that we should vote for an empty layer if committee signed more than one block intentionally, since it clearly malicious behavior. i believe that Tal mentioned it some time in the past, but i am not tracking that

Reply:
ok. i focused on your quotes from Tal and missed the malicious behavior.

filed this issue https://github.com/spacemeshos/go-spacemesh/issues/3467 to follow up.
@dshulyak @pigmej